packages/drivekit_trip_analysis/flutter_drivekit_trip_analysis_android/lib/src/adapter.dart

```diff
@@ -27,3 +27,466 @@ extension PigeonAdapter on Vehicle {
+      engineTempStatus: engineTempStatus,
+      coldFuelVolume: coldFuelVolume,
+    );
+  }
+}
+
+/// Adapts the [Safety] class to the corresponding Pigeon
+extension SafetyAdapter on Safety {
+  /// Converts a [PigeonSafety] to a corresponding Pigeon
+  PigeonSafety toPigeonImplementation() {
+    return PigeonSafety(
+      nbAdh: nbAdh,
+      nbAccel: nbAccel,
+      nbDecel: nbDecel,
+      nbAdhCrit: nbAdhCrit,
+      nbAccelCrit: nbAccelCrit,
+      nbDecelCrit: nbDecelCrit,
+      safetyScore: safetyScore,
+    );
+  }
+}
+
+/// Adapts the [ItineraryData] class to the corresponding Pigeon
+extension ItineraryDataAdapter on ItineraryData {
+  /// Converts a [ItineraryData] to a corresponding model class.
+  PigeonItineraryData toPigeonImplementation() {
+    return PigeonItineraryData(
+      startDate: startDate,
+      endDate: endDate,
+      departureCity: departureCity,
+      arrivalCity: arrivalCity,
+      departureAddress: departureAddress,
+      arrivalAddress: arrivalAddress,
+    );
+  }
+}
+
+/// Adapts the [Logbook] class to the corresponding Pigeon
+extension LogbookAdapter on Logbook {
+  /// Converts a [Logbook] to a corresponding Pigeon
+  PigeonLogbook toPigeonImplementation() {
+    return PigeonLogbook(
+      status: status,
+      updateDate: updateDate,
+    );
+  }
+}
+
+/// Adapts the [SpeedingStatistics] class to the Pigeon
+extension SpeedingStatisticsAdapter on SpeedingStatistics {
+  /// Converts a [PigeonSpeedingStatistics] to a corresponding Pigeon
+  PigeonSpeedingStatistics toPigeonImplementation() {
+    return PigeonSpeedingStatistics(
+      distance: distance,
+      duration: duration,
+      speedingDistance: speedingDistance,
+      speedingDuration: speedingDuration,
+      score: score,
+      speedLimitContexts: speedLimitContexts
+          .map((SpeedLimitContext e) => e.toPigeonImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedSafety] class to the corresponding Pigeon
+extension AdvancedSafetyAdapter on AdvancedSafety {
+  /// Converts a [PigeonAdvancedSafety] to a corresponding model class.
+  PigeonAdvancedSafety toPigeonImplementation() {
+    return PigeonAdvancedSafety(
+      safetyContext: safetyContext
+          .map((SafetyContext e) => e.toPigeonImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedEcoDriving] class to the corresponding Pigeon
+extension AdvancedEcoDrivingAdapter on AdvancedEcoDriving {
+  /// Converts a [AdvancedEcoDriving] to a corresponding Pigeon
+  PigeonAdvancedEcoDriving toPigeonImplementation() {
+    return PigeonAdvancedEcoDriving(
+      ecoDrivingContext:
+          ecoDrivingContext.map((e) => e.toPigeonImplementation()).toList(),
+    );
+  }
+}
+
+/// Adapts the [AdvancedFuelEstimation] class to the corresponding Pigeon
+extension AdvancedFuelEstimationAdapter on AdvancedFuelEstimation {
+  /// Converts a [PigeonAdvancedFuelEstimation] to a corresponding model class.
```

Comment:
"/// Converts a [**AdvancedFuelEstimation**] to a corresponding **Pigeon**"

Reply:
corrigé sur 7cecc2affa21960d1b1c86074dea479cc4822f35